src/api/settings/getServerSettings.js

```diff
@@ -44,5 +61,21 @@ export type ApiResponseServerSettings = {|
 |};
 
 /** See https://zulip.com/api/get-server-settings */
-export default async (realm: URL): Promise<ApiResponseServerSettings> =>
-  apiGet({ apiKey: '', email: '', realm }, 'server_settings');
+export default async (realm: URL): Promise<ApiResponseServerSettings> => {
+  const result = await apiGet({ apiKey: '', email: '', realm }, 'server_settings');
+  if (result.realm_name == null) {
```

Comment:
This means that the object this function actually returns always does have a `realm_name`, right?

Let's express that in the return type. That will let this function more fully act as a "crunchy shell", avoiding the need for the changes in other files.

We can separately express the type of what we actually expect from the server, following the pattern of other functions like `api.getMessages`, like this:
```js
type ServerApiResponseServerSettings = {|
  ...ApiResponseServerSettings,

  // … comment …
  realm_name?: string,
|};
```

Then say `const result: ServerApiResponseServerSettings = await apiGet(…);`, and Flow can check the types from there.